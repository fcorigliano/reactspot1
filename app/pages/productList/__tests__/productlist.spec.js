const React = require('react');
const ProdListView = require('../view');
const { render, screen, act, fireEvent, waitFor, within } = require('@testing-library/react');
require('core-js');
const restClient = require('nordic/restclient');
// jest.mock('nordic/restclient');

const { mockGet } = restClient;



describe('La view de Home', () => {
    let component; 
    let prodMock = [{id:'MLA12874', title:'Samsung', thumbnail:'foto', permalink:'link'}]

    beforeEach(async() => {  
        await act(async() =>{
            mockGet.mockResolvedValueOnce({data: [{id:'MLA12874', title:'Moto', thumbnail:'foto', permalink:'link'}]})
            component= await waitFor(() => render(<ProdListView prodlist={prodMock} />));
        })
    });
    
    it('1) Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    
    it('2) Debe modificar la lista de productos al hacer click en el boton', async() => {
        await act(async () => {
            const button = await screen.findByRole('pagination');
            fireEvent.click(button);
        })
        
        const title = await screen.findByText('Moto')
        expect(title).toBeInTheDocument();
    })

    it('Opcional: Si no hay productos deberia mostrar el msj "no se encontraron productos"', async() => {
        await act(async() =>{
            component= await waitFor(() => render(<ProdListView prodlist={[]}  />));
        })
        const text = screen.getByText(/no se encontraron productos/i)
        expect(text).toBeInTheDocument();
    })

})