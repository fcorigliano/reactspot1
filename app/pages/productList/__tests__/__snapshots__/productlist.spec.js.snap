// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`La view de Home 1) Renderiza 1`] = `
<DocumentFragment>
  <div
    class="demo"
  >
    <h1>
      productList
    </h1>
    <button
      role="pagination"
    >
      NEXT
    </button>
    <ol>
      <li
        class="card"
      >
        <figure
          class="img"
        >
          <img
            alt="Samsung"
            class=""
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          />
        </figure>
        <div
          class="info-products"
        >
          <h4
            class="price"
          >
            $
          </h4>
          <h3
            class="title-product"
          >
            Samsung 
          </h3>
        </div>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
